src/movies.js

```diff
@@ -1,16 +1,94 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies) {
+	const directors = movies.map(key => key.director);
+	return directors;
+}
+getAllDirectors(movies);
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function removeDuplicates(originalArray, prop) {
+	let newArray = [];
+	let lookupObject  = {};
+
+	for(let i in originalArray) {
+		 lookupObject[originalArray[i][prop]] = originalArray[i];
+	}
+
+	for(i in lookupObject) {
+			newArray.push(lookupObject[i]);
+	}
+	 return newArray;
+}
+
+let uniqueArray = removeDuplicates(movies, "director");
+console.log("uniqueArray is: " + JSON.stringify(uniqueArray));
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+	const stevenDrama = movies.filter(key => key.director === "Steven Spielberg").filter(key => key.genre.includes("Drama")).length;
+	return stevenDrama;
+};
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
-// Iteration 4: Drama movies - Get the average of Drama Movies
+function ratesAverage(movies) {
+	const totalMovies = movies.length;
+	let sumAverage = movies.reduce((acc, key) => {
+		if (key.rate) {
+			return acc + key.rate
+		} else {
+			return acc
+		}
+	}, 0)
+		if (movies.length === 0) {
+			return 0;
+		}
+			return Math.round((sumAverage / totalMovies) * 100) / 100;
+}
+ratesAverage(movies);
+
+// Iteration 4: Drama movies - Get the average of Drama er(key) Movies
+
+function dramaMoviesRate(movies) {
+	const dramaMovies = movies.filter(key => key.genre.includes("Drama")).filter(key => key.rate)
+	if (dramaMovies.length === 0) {
+		return 0;
+	}
+	const averageRateDrama = dramaMovies.reduce((acc, key) => acc + key.rate, 0) / dramaMovies.length;
```

Comment:
Plutôt que de coder à nouveau average, vous pouvez utiliser la function de l'itération 3